controllers/configurationpolicy_utils.go

```diff
@@ -115,6 +115,10 @@ func equalObjWithSort(mergedObj interface{}, oldObj interface{}) (areEqual bool)
 		// this includes the case where oldObj is nil
 		return false
 	case []interface{}:
+		if len(mergedObj) == 0 && oldObj == nil {
```

Comment:
On what was line 170, `case []map[string]interface{}`, the condition here is not necessary, because the length would never be 0. If it was length zero, it would just be `[]`, and that doesn't have the required type information to get into that case.

I think the rest of the logic in `checkFieldsWithSort` should already cover these cases. At least, that was the intention of https://github.com/open-cluster-management-io/config-policy-controller/commit/a663462ac9c812bb87b6857abf1e40e59c3c4397, and what I was thinking of adding seems equivalent.